dbg.h

```diff
@@ -431,23 +438,47 @@ inline bool pretty_print(std::ostream& stream, const std::tuple<>&) {
   return true;
 }
 
+// Converts decimal integer to binary string
+std::string decimalToBinary(unsigned n) {
+  std::string toRet;
+  do {
+    toRet.push_back(n % 2 + '0');
+    n /= 2;
+  } while (n != 0);
+
+  std::reverse(toRet.begin(), toRet.end());
+  return toRet;
+}
+
```

Comment:
I played around with this for a while. Here is a version with a few improvements (I think):

* avoids `std::reverse`
* has zero-padding (like we have for 
* works for arbitrary integer types

```c++
template <typename T>
std::string decimalToBinary(T n) {
  const auto length = 8 * sizeof(T);
  std::string result;
  result.resize(length);

  for (int i = 0; i < length; ++i) {
    const auto bit_at_index_i = (n >> i) & 1;
    result[length - 1 - i] = bit_at_index_i + '0';
  }

  return result;
}
```

What do you think?

Reply:
I don't think this is the best. Main problem I see is :   
`long int x = 54; dbg(dbg:bin(x));` yeilds `0b0000000000000000000000000000000000000000000000000000000000110110`   

That's not very nice. Using the sizeof function does not work unless we trim off leading zeros which is more work than necessary. We could check if the string is under, say, 4 characters and then add some leading zeros.

The other method is using logs to determine how many characters are needed but then it looks something like this. :  
```c++
if(n == 0)
    return std::string("0");
const auto length = (int)std::floor(log2(n)) + 1;
```
Which does not appeal to me.  
Though I can change the  current function to be a template and will do that seeing as that is nicer.

Reply:
Just thought of something else. Now the function uses insert to add the new number to the beginning of the string. See ea0eb83

Reply:
> That's not very nice. Using the sizeof function does not work unless we trim off leading zeros which is more work than necessary.

Adding the zero padding was intended as a feature, not as a an unfortunate side-effect. We use zero padding for the other bases as well. Yes, it is very long for 64-bit integers, but if a user really has a bitmap that is 64 bits long, most of these bits will probably not be zero anyway.

> Just thought of something else. Now the function uses insert to add the new number to the beginning of the string. See [ea0eb83](https://github.com/sharkdp/dbg-macro/commit/ea0eb834e4a443e5b2a1b7e94af534024052f44f)

I don't think that's a good alternative. Now the algorithm is `O(n²)` in the number of bits because `insert` is `O(n)` for `std::string` (it's not really specified in the standard, but it's "likely linear").

We are not too concerned about performance here, but you can see that both of your implementations are significantly slower: http://quick-bench.com/Cj14RHAgmFcqK-6WprINWeHf5TM

![image](https://user-images.githubusercontent.com/4209276/71806967-81c41780-306a-11ea-999a-633764d60163.png)

Reply:
Well, fair enough points. I'll switch it over.